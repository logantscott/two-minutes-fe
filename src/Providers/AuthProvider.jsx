/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { AuthContext } from '../hooks/AuthContext';
import { fetchSignup, fetchLogin } from '../services/auth';

const AuthProvider = ({ children }) => {
  const [currentUser, setCurrentUser] = useState(null);
  const [authError, setAuthError] = useState(null);

  const authService = (serviceFn, ...args) => {
    setAuthError(null);
    return serviceFn(...args)
      .then(user => setCurrentUser(user))
      .catch(err => setAuthError(err));
  };

  const signup = (name, email, password) =>
    authService(fetchSignup, name, email, password);

  const login = (email, password) => 
    authService(fetchLogin, email, password);

  const signout = () => {
    setCurrentUser(null);
  };

  return (
    <AuthContext.Provider value={{ currentUser, signup, login, signout }}>
      {children}
    </AuthContext.Provider>
  );
};

export default AuthProvider;
