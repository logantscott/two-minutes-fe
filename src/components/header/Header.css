.Header {
  li {
    list-style: none;
    display: inline-block;
    margin-right: 8px;
  }
  a {
    text-decoration: underline;
    color: blue;
    cursor: pointer;
  }
}