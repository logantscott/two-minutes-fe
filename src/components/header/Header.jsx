import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.css';
import { useCurrentUser } from '../../hooks/AuthContext';

const Header = () => {
  const currentUser = useCurrentUser();

  return (
    <nav className={styles.Header}>
      <ul>
        <li><Link to='/'>Home</Link></li>
        <li>{
          currentUser ?
            <>
              <Link to='/dashboard'>Dashboard</Link>
              <Link to='/signout'>Sign Out</Link> 
            </>
            : <Link to='/login'>Signup/Login</Link>
        }</li>
      </ul>
    </nav>
  );
};

export default Header;
