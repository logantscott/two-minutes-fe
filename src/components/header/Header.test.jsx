import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import AuthProvider from '../../Providers/AuthProvider';

describe('Header component', () => {
  it('renders Header', () => {
    const wrapper = shallow(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
