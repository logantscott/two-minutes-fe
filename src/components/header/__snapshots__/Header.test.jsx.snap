// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<ContextProvider
  value={
    Object {
      "authError": null,
      "authLoading": true,
      "currentUser": null,
      "login": [Function],
      "logout": [Function],
      "signup": [Function],
    }
  }
>
  <Header />
</ContextProvider>
`;
